torch/cuda/random.py

```diff
@@ -64,6 +65,30 @@ def cb():
     _lazy_call(cb)
 
 
+def set_rng_state_offset(offset: int, device: Union[int, str, torch.device] = 'cuda') -> None:
+    r"""Sets the random number generator state offset of the specified GPU.
+
+    Args:
+        offset (int): The desired offset
+        device (torch.device or int, optional): The device to set the RNG state.
+            Default: ``'cuda'`` (i.e., ``torch.device('cuda')``, the current CUDA device).
+    """
+    if isinstance(device, str):
+        device = torch.device(device)
+    elif isinstance(device, int):
+        device = torch.device('cuda', device)
+
+    def cb():
+        idx = cast(torch.device, device).index
+        if idx is None:
+            idx = current_device()
+        default_generator = torch.cuda.default_generators[idx]
```

Comment:
This is very copy-pastey with the function above. Refactor?

Reply:
Done